fix: use more secure hash function

Piccirello opened the pull request:
This cache key holds user input, so we want stronger collision resistance. This change will invalidate the existing cache.

Will deploy this over the weekend when the cache is the least used.

Piccirello committed:
Use more secure hash function

These cache keys are controlled by user input, so we want stronger collision resistance. This change will invalidate the existing caches.

Piccirello committed:
Update hash in test

rafaeelaudibert commented on posthog/caching/login_device_cache.py:
```diff
@@ -10,8 +10,7 @@ def check_and_cache_login_device(user_id: int, location: str, short_user_agent:
     device_fingerprint = f"{location}:{short_user_agent}"
-    # nosemgrep: python.lang.security.insecure-hash-algorithms-md5.insecure-hash-algorithm-md5
-    device_hash = hashlib.md5(device_fingerprint.encode()).hexdigest()
+    device_hash = hashlib.sha256(device_fingerprint.encode()).hexdigest()
```
Not on the team myself, but people kinda complain about this sometimes, maybe we should disable sending the emails while we backfill the cache - like 7 days or smth?

rafaeelaudibert reviewed (approved):
Anything after friday 10pm pst  is easy here, Ive done on Saturday mornings without a blip

Piccirello committed:
Include team pk in cache key

This makes it easier to invalidate a specific team's cache. It also massively reduces the blast radius of a hash conflict (which should be infeasible anyway with sha256).

Piccirello committed:
Revert login device fingerprint change

Give us more time to figure our rollout strategy while shipping the rest.

Piccirello committed:
I love python types they're really helpful
